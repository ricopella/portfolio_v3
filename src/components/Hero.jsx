import styled from '@emotion/styled'
import { graphql, useStaticQuery } from 'gatsby'
import BackgroundImage from 'gatsby-background-image'
import React from 'react'

const ImageBackground = styled(BackgroundImage)`
  background-size: cover;
  height: 100vh;
  position: 'unset';
  opacity: 0.95;
  max-width: 100%;
  padding: 0;
  grid-row: 1/-1;
  grid-column: 1/-1;
  min-height: 100vh;

  + * {
    margin-top: 0;
  }
`

const Hero = () => {
  const { image } = useStaticQuery(graphql`
    query {
      image: file(relativePath: { eq: "joseph-ngabo.jpg" }) {
        sharp: childImageSharp {
          fluid(maxWidth: 2000, quality: 100) {
            ...GatsbyImageSharpFluid
            presentationWidth
          }
        }
      }
    }
  `)
  return (
    <ImageBackground tag="section" fluid={image.sharp.fluid} fadeIn="soft" />
  )
}

export default Hero
