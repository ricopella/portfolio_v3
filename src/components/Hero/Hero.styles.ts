import BackgroundImage from 'gatsby-background-image'
import styled from '@emotion/styled'
import { COLORS } from '../../styles/variables'

const BackgroundGradient = styled.div`
  background: -webkit-linear-gradient(
    360deg,
    ${props => props.theme.backgroundColor} 0.5%,
    ${props => props.theme.actionColor} 360%
  );
  background: linear-gradient(
    360deg,
    ${props => props.theme.backgroundColor} 0.5%,
    ${props => props.theme.actionColor} 360%
  );
  grid-column: 1/-1;
  grid-row: 1/-1;
  height: 100vh;
  max-width: 100%;
  min-height: 100vh;
`

export default BackgroundGradient
