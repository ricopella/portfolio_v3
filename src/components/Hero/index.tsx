import BackgroundGradient from './Hero.styles'
import React from 'react'

const Hero = () => <BackgroundGradient />

export default Hero
