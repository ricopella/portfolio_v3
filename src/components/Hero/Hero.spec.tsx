import Hero from './index'
import React from 'react'
import renderer from 'react-test-renderer'

describe('<Hero />', () => {
  it('should match Hero snapshot', () => {
    const internalTree = renderer.create(<Hero />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })
})
