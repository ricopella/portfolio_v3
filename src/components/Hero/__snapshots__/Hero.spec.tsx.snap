// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match Hero snapshot 1`] = `
<div
  className="css-1wwn65t"
/>
`;
